libraries/chain/db_maint.cpp

```diff
@@ -800,6 +800,42 @@ void update_and_match_call_orders( database& db )
    wlog( "Done updating all call orders for hardfork core-343 at block ${n}", ("n",db.head_block_num()) );
 }
 
+void database::process_bitassets()
+{
+   time_point_sec head_time = head_block_time();
+   uint32_t head_epoch_seconds = head_time.sec_since_epoch();
+   bool after_hf_core_518 = ( head_time >= HARDFORK_CORE_518_TIME ); // clear expired feeds
+
+   for( const auto& d : get_index_type<asset_bitasset_data_index>().indices() )
+   {
+      modify( d, [this,head_time,head_epoch_seconds,after_hf_core_518]( asset_bitasset_data_object &o )
+      {
+         o.force_settled_volume = 0; // Reset all BitAsset force settlement volumes to zero
+
+         // clear expired feeds
+         if( after_hf_core_518 )
+         {
+            const auto &asset = get( o.asset_id );
+            auto flags = asset.options.flags;
+            if ( ( flags & ( witness_fed_asset | committee_fed_asset ) ) &&
+                 o.options.feed_lifetime_sec < head_epoch_seconds ) // if smartcoin && check overflow
+            {
+               fc::time_point_sec calculated = head_time - o.options.feed_lifetime_sec;
+               for( auto itr = o.feeds.rbegin(); itr != o.feeds.rend(); ) // loop feeds
+               {
+                  auto feed_time = itr->second.first;
+                  std::advance( itr, 1 );
+                  if( feed_time < calculated )
+                     o.feeds.erase( itr.base() ); // delete expired feed
+               }
```

Comment:
Up to you, but you could replace this with something like:

    o.feeds.erase( 
        std::remove_if(o.feeds.begin(), o.feeds.end(),
            [calculated] (auto f) { return f.second.first < calculated; }),
        o.feeds.end());

Reply:
Oops, sorry. remove_if doesn't work for map. Please ignore